UM/Application.py

```diff
@@ -348,4 +357,57 @@ def getInstallPrefix():
         else:
             return os.path.abspath(os.path.join(os.path.dirname(sys.executable), ".."))
 
+    def _mrproper(self, only_user = False):
+        path_list = []
+
+        if not only_user:
+            if Platform.isWindows():
+                os.path.join(os.path.expanduser(), )
+                dir_patterns = ["\\AppData\\Local\\%s" %self._application_name,
+                                ]
+                file_patterns = []
+            elif Platform.isOSX():
+                dir_patterns = ["Library/Application Support/%s" %self._application_name,
+                                ]
+                file_patterns = []
+            elif Platform.isLinux():
+                dir_patterns = ['.config/%s' %self._application_name,
+                                '.cache/%s' %self._application_name,
+                                '.local/share/%s' %self._application_name,
+                                ]
+                file_patterns = ['.config/%src' %self._application_name,
+                                 ]
+            else:
+                print("Unknown platform. Nothing to delete!")
+                return
+
+            user_basedir = os.path.split(os.path.expanduser("~"))[0]
+            user_paths = [os.path.join(user_basedir, entry) for entry in os.listdir(user_basedir)]
+            for user_path in user_paths:
+                for dir_pattern in dir_patterns:
+                    path_list.append(os.path.join(user_path, dir_pattern))
+                for file_pattern in file_patterns:
+                    path_list.append(os.path.join(user_path, file_pattern))
+
+        else:
+            if Platform.isLinux() or Platform.isOSX() or Platform.isWindows():
```

Comment:
Unnecessary checking.

Reply:
Hmm, when looking at that I guess I had in mind to catch the case when Cura is running on none of our known OSs. But as soon as we are using `Resources` I think this is nothing that needs to get checked.